File: 0113-path-sum-ii/0113-path-sum-ii.java
class Solution {
    public void helper(TreeNode  root,List<List<Integer>> ans,List<Integer> ls,int sum){
        if(root==null) return;
         ls.add(root.val);
         if(root.right==null && root.left==null){
            if(sum==root.val){

                 ans.add(new ArrayList<>(ls));
                // return;
            }
           
         }

         helper(root.left,ans,ls,sum-root.val);
         helper(root.right,ans,ls,sum-root.val);
         ls.remove(ls.size()-1)
;    }
    public List<List<Integer>> pathSum(TreeNode root, int targetSum) {
      List<List<Integer>> ans=new ArrayList<>(); 
    //   List<Integer> ls=new ArrayList<>();
       helper(root,ans,new ArrayList<>(),targetSum);    
       return ans;
     }
}

File: 0113-path-sum-ii/0113-path-sum-ii.cpp

class Solution {
public:
    void helper(TreeNode* root,vector<int> v,vector<vector<int>> &ans,int sum){
        if(!root) return ;
        if(!root->right && !root->left){
            if(root->val==sum){
            v.push_back(root->val);
            ans.push_back(v);
            return;
            }
           else{
            return;
           }
        }
        v.push_back(root->val);
        helper(root->left,v,ans,sum-(root->val));
        helper(root->right,v,ans,sum-(root->val));
        v.pop_back();
    }
    vector<vector<int>> pathSum(TreeNode* root, int targetSum) {
         vector<int> v;
         vector<vector<int>> ans;
         helper(root,v,ans,targetSum);
         return ans;

    }
};